File: tests/fixtures/ioredis-app/src/cache.ts
// @ts-nocheck
import { redis } from './client';

export async function getUser(id: string) {
  const cached = await redis.get(`user:${id}`);
  if (cached) return JSON.parse(cached);
  return null;
}

export async function setUser(id: string, data: unknown) {
  await redis.set(`user:${id}`, JSON.stringify(data));
  await redis.expire(`user:${id}`, 3600);
}

export async function incrViews(id: string) {
  return redis.incr(`views:${id}`);
}

File: tests/fixtures/redis-py-app/cache.py
import json

from client import r


def get_user(uid: str) -> dict | None:
    cached = r.get(f"user:{uid}")
    if cached:
        return json.loads(cached)
    return None


def set_user(uid: str, data: dict) -> None:
    r.set(f"user:{uid}", json.dumps(data))
    r.expire(f"user:{uid}", 3600)


def incr_views(uid: str) -> int:
    return r.incr(f"views:{uid}")
